core/src/test/java/com/graphhopper/routing/QueryGraphTest.java

```diff
@@ -613,21 +613,46 @@ public void testEnforceHeadingByEdgeId() {
         queryGraph.lookup(Arrays.asList(qr));
 
         // enforce coming in north
-        queryGraph.enforceHeadingByEdgeId(2, 1, false);
+        queryGraph.enforceHeadingByEdgeId(2, 1, true);
         // test penalized south
-        boolean expect = true;
         VirtualEdgeIteratorState incomingEdge = (VirtualEdgeIteratorState) queryGraph.getEdgeIteratorState(1, 2);
+        VirtualEdgeIteratorState incomingEdgeReverse = (VirtualEdgeIteratorState) queryGraph.getEdgeIteratorState(1, incomingEdge.getBaseNode());
+        boolean expect = true;  // expect incoming edge to be avoided
+        assertEquals(expect, incomingEdge.getBool(EdgeIteratorState.K_UNFAVORED_EDGE, !expect));
+        expect = false;         // expect reverse incoming edge not to be avoided
```

Comment:
Why not remove `expect = false`. That way it would fail before your PR => Showing the issue. Then you add your PR and therefore fix the failing test case. The order is not important at this point :).